Packet++/header/HttpLayer.h

```diff
@@ -207,200 +207,280 @@ namespace pcpp
-			Http505HTTPVersionNotSupported,
+			Http505HTTPVersionNotSupported = 505,
 			/** 506 Variant Also Negotiates */
-			Http506VariantAlsoNegotiates,
+			Http506VariantAlsoNegotiates = 506,
 			/** 507 Insufficient Storage */
-			Http507InsufficientStorage,
+			Http507InsufficientStorage = 507,
 			/** 508 Loop Detected */
-			Http508LoopDetected,
+			Http508LoopDetected = 508,
 			/** 509 Bandwidth Limit Exceeded */
-			Http509BandwidthLimitExceeded,
+			Http509BandwidthLimitExceeded = 509,
 			/** 510 Not Extended */
-			Http510NotExtended,
+			Http510NotExtended = 510,
 			/** 511 Network Authentication Required */
-			Http511NetworkAuthenticationRequired,
+			Http511NetworkAuthenticationRequired = 511,
+			/** 512-519 unassigned **/
 			/** 520 Origin Error */
-			Http520OriginError,
+			Http520OriginError = 520,
 			/** 521 Web server is down */
-			Http521WebServerIsDown,
+			Http521WebServerIsDown = 521,
 			/** 522 Connection timed out */
-			Http522ConnectionTimedOut,
+			Http522ConnectionTimedOut = 522,
 			/** 523 Proxy Declined Request */
-			Http523ProxyDeclinedRequest,
+			Http523ProxyDeclinedRequest = 523,
 			/** 524 A timeout occurred */
-			Http524aTimeoutOccurred,
+			Http524aTimeoutOccurred = 524,
+			/** 525-597 unassigned **/
 			/** 598 Network read timeout error */
-			Http598NetworkReadTimeoutError,
+			Http598NetworkReadTimeoutError = 598,
 			/** 599 Network connect timeout error */
-			Http599NetworkConnectTimeoutError,
+			Http599NetworkConnectTimeoutError = 599,
+
 			/** Unknown status code */
-			HttpStatusCodeUnknown
+			HttpStatus1xxCodeUnknown = 900001, // 1xx: Informational - Request received, continuing process
+			HttpStatus2xxCodeUnknown = 900002, // 2xx: Success - The action was successfully received, understood, and accepted
+			HttpStatus3xxCodeUnknown = 900003, // 3xx: Redirection - Further action must be taken in order to complete the request
+			HttpStatus4xxCodeUnknown = 900004, // 4xx: Client Error - The request contains bad syntax or cannot be fulfilled
+			HttpStatus5xxCodeUnknown = 900005, // 5xx: Server Error - The server failed to fulfill an apparently valid request
+			HttpStatusCodeError = 999999, // other arbitrary number
+		};
+
+		HttpResponseStatusCode() = default;
+		// cppcheck-suppress noExplicitConstructor
+		constexpr HttpResponseStatusCode(Value statusCode) : m_value(statusCode) { }
+
+		constexpr operator Value() const { return m_value; }
+		explicit operator bool() const = delete;
+
+		/**
+	 	 * @struct HttpResponseStatusCodeHash
+		 * @brief The helper structure for hash HttpResponseStatusCode while using std::unordered_map
+		 */
+		struct HttpResponseStatusCodeHash {
+			size_t operator()(const HttpResponseStatusCode& status) const {
+				return std::hash<int>()(static_cast<int>(status));
+			}
 		};
 
+		/**
+		 * @brief A pointer to the first line instance for this message
+		 */
+		std::string toString() const {
+			return std::to_string(m_value);
+		}
```

Comment:
I'm wondering if we also want to expose the code message and not only the value.
Currently this method will return the code as string, for example: `"200"` but there's no method to return `"OK"`.
If we expose a method to return the message, we can also support customer messages. For example, you can do:
```cpp
HttpResponseStatusCode statusCode1(HttpResponseStatusCode::Http200OK);
HttpResponseStatusCode statusCode2(HttpResponseStatusCode::Http200OK, "Custom message");

std::cout << statusCode1.getMessage();
// prints "OK"
std::cout << statusCode2.getMessage();
// prints "Custom message"
```

But maybe we can leave it to a separate PR?

Reply:
I like the idea. let's add this `getMessage` function in this PR